pkg/policy/k8s/watcher.go

```diff
@@ -44,11 +47,19 @@ type policyWatcher struct {
 	// avoid key clashing between CNPs and CCNPs.
 	// The cache contains CNPs and CCNPs in their "original form"
 	// (i.e: pre-translation of each CIDRGroupRef to a CIDRSet).
-	cnpCache       map[resource.Key]*types.SlimCNP
+	cnpCache map[resource.Key]*types.SlimCNP
+
 	cidrGroupCache map[string]*cilium_api_v2alpha1.CiliumCIDRGroup
-	// cidrGroupPolicies is the set of policies that are referencing CiliumCIDRGroup objects.
-	cidrGroupPolicies map[resource.Key]struct{}
-	// cidrGroupPolicies is the set of policies that contain ToServices references
+
+	// cidrGroupCIDRs is the set of CIDRs upserted in to the ipcache
+	// for a given cidrgroup
+	cidrGroupCIDRs map[string]sets.Set[netip.Prefix]
+
+	// cidrGroupRefCount is the number of policies that reference a given
+	// cidr group. Groups with no references may not be inserted in to the ipcache.
+	cidrGroupRefCount map[string]int
```

Comment:
nit: You could probably use the `pkg/counter` here, this would save you a few lines of code